From d59d2adaa479f55afc02651caed15a6c1538d681 Mon Sep 17 00:00:00 2001
From: Dave Gerlach <d-gerlach@ti.com>
Date: Thu, 5 Jan 2017 15:32:22 -0600
Subject: [PATCH 661/895] soc: ti: wkup_m3_ipc: Fix misassigned struct device
 pointer

The firmware loading callback for the volt scaling firmware expects the
the second argument to be a pointer to a struct wkup_m3_ipc, however the
function also attempts to cast it to be used directly as a struct device
pointer. This incorrect pointer is only used for dev_err messages so if
a failure occurs it will result in a crash.

Instead let's use the struct device pointer inside the struct
wkup_m3_ipc pointer that is actually passed so that we have a valid
dev pointer.

Fixes: 1be271922084 ("soc: ti: wkup_m3_ipc: Add support for i2c voltage
scaling")
Signed-off-by: Dave Gerlach <d-gerlach@ti.com>
Acked-by: Suman Anna <s-anna@ti.com>

(cherry picked from commit 19694ee14597e42681a9b8ed74c5f0ce338d73f4)
---
 drivers/soc/ti/wkup_m3_ipc.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/soc/ti/wkup_m3_ipc.c b/drivers/soc/ti/wkup_m3_ipc.c
index d033765..11c449f 100644
--- a/drivers/soc/ti/wkup_m3_ipc.c
+++ b/drivers/soc/ti/wkup_m3_ipc.c
@@ -122,7 +122,7 @@ static void wkup_m3_scale_data_fw_cb(const struct firmware *fw, void *context)
 	unsigned long val, aux_base;
 	struct wkup_m3_scale_data_header hdr;
 	struct wkup_m3_ipc *m3_ipc = context;
-	struct device *dev = (struct device *)context;
+	struct device *dev = m3_ipc->dev;
 
 	if (!fw) {
 		dev_err(dev, "Voltage scale fw name given but file missing.\n");
-- 
1.7.5.4

